Tests/AriesFrameworkTests/connection-less/ConnectionlessExchangeTest.swift

```diff
@@ -0,0 +1,119 @@
+        try await holderAgent.initialize()
+
+        credDefId = try await TestHelper.prepareForIssuance(issuerAgent, ["name", "age"])
+    }
+
+    override func tearDown() async throws {
+        try await issuerAgent?.reset()
+        try await holderAgent?.reset()
+        try await super.tearDown()
+    }
+
+    func validateState(for agent: Agent, threadId: String, state: CredentialState) async throws {
+        let record = try await agent.credentialExchangeRepository.getByThreadAndConnectionId(threadId: threadId, connectionId: nil)
+        XCTAssertEqual(record.state, state, "agent=\(agent.agentConfig.label)")
+    }
+
+    func validateState(for agent: Agent, threadId: String, state: ProofState) async throws {
+        let record = try await agent.proofRepository.getByThreadAndConnectionId(threadId: threadId, connectionId: nil)
+        XCTAssertEqual(record.state, state, "agent=\(agent.agentConfig.label)")
+    }
+
+    func testConnectionlessCredentialExchange() async throws {
+        let offerOptions = CreateOfferOptions(
+            connection: faberConnection,
+            credentialDefinitionId: credDefId,
+            attributes: credentialPreview.attributes,
+            comment: "this is credential-offer for you")
+        let (message, record) = try await issuerAgent.credentialService.createOffer(options: offerOptions)
+        try await validateState(for: issuerAgent, threadId: record.threadId, state: CredentialState.OfferSent)
+
+        let oobConfig = CreateOutOfBandInvitationConfig(
+            label: "issuer-to-holder-invitation",
+            alias: "issuer-to-holder-invitation",
+            imageUrl: nil,
+            goalCode: nil,
+            goal: nil,
+            handshake: false,
+            messages: [message],
+            multiUseInvitation: false,
+            autoAcceptConnection: true,
+            routing: nil)
+        let oobInvitation = try await issuerAgent.oob.createInvitation(config: oobConfig)
+
+        var (oob, connection) = try await holderAgent.oob.receiveInvitation(oobInvitation.outOfBandInvitation)
+        XCTAssertNotNil(connection)
+        XCTAssertEqual(connection?.state, .Complete) // this is a fake connection.
+        XCTAssertNotNil(oob)
+
+        (oob, connection) = try await holderAgent.oob.acceptInvitation(outOfBandId: oob.id, config: receiveInvitationConfig)
+        XCTAssertNotNil(connection)
+        XCTAssertNotNil(oob)
+        try await validateState(for: holderAgent, threadId: record.threadId, state: CredentialState.RequestSent)
+
+        try await Task.sleep(nanoseconds: UInt64(5 * SECOND))
+
+        try await validateState(for: holderAgent, threadId: record.threadId, state: CredentialState.Done)
+        try await validateState(for: issuerAgent, threadId: record.threadId, state: CredentialState.Done)
+    }
+
+    func validateProofExchangeRecordState(for agent: Agent, threadId: String, state: ProofState) async throws {
```

Comment:
There's the same function defined above with the name `validateState()`.

Reply:
It's kind of function overloading for readability of test code. Would it be better to separate function names?

Reply:
I don't have any preference. It will be okay to keep only one function of the two.